bindings/cs/MongoDB.Libmongocrypt/HashCallback.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * Copyright 2020–present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Security.Cryptography;
+
+namespace MongoDB.Libmongocrypt
+{
+    internal static class HashCallback
+    {
+        public static bool Hash(
+            IntPtr ctx,
+            IntPtr @in,
+            IntPtr @out,
+            IntPtr statusPtr)
+        {
+            var status = new Status(StatusSafeHandle.FromIntPtr(statusPtr));
```

Comment:
`StatusSafeHandle` derives from `SafeHandle`, which is both `IDisposable` and has a finalizer. If we don't dispose of the `SafeHandle` explicitly, then we force the runtime to go down the finalizer route to clean up the associated resources. We should put this in a `using` block.